static/app/components/issueList.tsx

```diff
@@ -23,155 +23,105 @@ type Props = WithRouterProps & {
-      this.remountComponent();
-    }
-  }
-
-  remountComponent() {
-    this.setState(this.getInitialState(), this.fetchData);
-  }
+}
 
-  fetchData = () => {
-    const {location, api, endpoint, query} = this.props;
+function IssueList({
+  api,
+  endpoint,
+  emptyText,
+  query,
+  location,
+  pagination,
+  renderEmpty,
+  noBorder,
+  noMargin,
+}: IssueListProps): React.ReactElement {
+  const [state, setState] = React.useState<IssueListState>({
+    issueIds: [],
+    status: 'loading',
+    pageLinks: null,
+    data: [],
+  });
+
+  const fetchIssueListData = React.useCallback(() => {
     api.clear();
     api.request(endpoint, {
       method: 'GET',
       query: {
-        cursor: (location && location.query && location.query.cursor) || '',
         ...query,
+        ...(location?.query?.cursor ? {cursor: location.query.cursor} : {}),
       },
       success: (data, _, resp) => {
-        this.setState({
+        setState({
           data,
-          loading: false,
-          error: false,
+          status: 'success',
           issueIds: data.map(item => item.id),
           pageLinks: resp?.getResponseHeader('Link') ?? null,
         });
       },
       error: () => {
-        this.setState({loading: false, error: true});
+        setState({...state, status: 'error'});
       },
     });
-  };
+  }, [query, endpoint, location.query, api]);
 
-  renderError() {
-    return (
-      <div style={{margin: `${space(2)} ${space(2)} 0`}}>
-        <LoadingError onRetry={this.fetchData} />
-      </div>
-    );
-  }
-
-  renderLoading() {
-    return (
-      <div style={{margin: '18px 18px 0'}}>
-        <LoadingIndicator />
-      </div>
-    );
-  }
-
-  renderEmpty() {
-    const {emptyText} = this.props;
-    const {noBorder, noMargin} = this.props;
-    const panelStyle: React.CSSProperties = noBorder ? {border: 0, borderRadius: 0} : {};
-    if (noMargin) {
-      panelStyle.marginBottom = 0;
-    }
-
-    return (
-      <Panel style={panelStyle}>
-        <EmptyMessage icon={<IconSearch size="xl" />}>
-          {emptyText ? emptyText : t('Nothing to show here, move along.')}
-        </EmptyMessage>
-      </Panel>
-    );
-  }
-
-  renderResults() {
-    const {noBorder, noMargin, renderEmpty} = this.props;
-    const {loading, error, issueIds, data} = this.state;
-
-    if (loading) {
-      return this.renderLoading();
+  React.useEffect(() => {
+    // TODO: location should always be passed as a prop, check why we have this
+    if (!location) {
+      return;
     }
 
-    if (error) {
-      return this.renderError();
-    }
+    setState({issueIds: [], status: 'loading', pageLinks: null, data: []});
+    fetchIssueListData();
+  }, [fetchIssueListData]);
 
-    if (issueIds.length > 0) {
-      const panelStyle: React.CSSProperties = noBorder
-        ? {border: 0, borderRadius: 0}
-        : {};
-      if (noMargin) {
-        panelStyle.marginBottom = 0;
-      }
+  const panelStyles = React.useMemo(() => {
```

Comment:
should we really use `useMemo` here?

Reply:
Yeah, I think in the context of react this is somewhat important because we are otherwise creating a new object reference each time IssueList rerenders which will cause the Panel component to rerender when it may not have been necessary for it to rerender